app/controllers/application_controller.rb

```diff
@@ -20,6 +21,15 @@ def after_sign_in_path_for(user)
     end
   end
 
+  def check_user_subdomain(user)
+    user_subdomain = user.organization.subdomain
+    if user_subdomain != request.subdomain
+      sign_out user
+      flash[:alert] = %Q[You are not registered with this subdomain, please sign in at <a href="http://#{user_subdomain}.digitallearn.org">#{user_subdomain}.digitallearn.org</a>]
+      root_path
```

Comment:
Also, one day, it would be good to setup devise to it doesn't care about your subdomain for authenticating, but then we just authorize based on subdomain and redirect without breaking session.